litmus-portal/backend/graphql-server/pkg/graphql/subscriptions/subscription.go

```diff
@@ -50,10 +50,10 @@ func SendWorkflowRequest(wfRequest *database.ChaosWorkFlowInput, r store.StateDa
 	}
 
 	r.Mutex.Lock()
-	if r.ClusterEventPublish != nil {
-		for _, observer := range r.ConnectedCluster {
-			observer <- newAction
-		}
+
+	if observer, ok := r.ConnectedCluster[wfRequest.ClusterID]; ok{
+		observer <- newAction
 	}
+
```

Comment:
We should send back an error to the mutation resolver if the `ok` is false i.e. when mentioned cluster is not connected

Reply:
Yes, this needs to be done. In this case, we have to insert data to DB after the request is successfully sent to the subscriber. Otherwise, useless data will be stored in the DB